backend/tracim_backend/lib/core/user.py

```diff
@@ -271,7 +272,33 @@ def set_email(
         )
         return user
 
-    def _check_email(self, email: str) -> bool:
+    def _check_email(self, email) -> bool:
+        """
+        Check if email is completely ok to be used in user db table
+        """
+        is_email_correct = self._check_email_correctness(email)
+        if not is_email_correct:
+            raise EmailValidationFailed(
+                'Email given form {} is uncorrect'.format(email))  # nopep8
+        email_already_exist_in_db = self.check_email_already_in_db(email)
+        if email_already_exist_in_db:
+            raise EmailAlreadyExistInDb(
+                'Email given {} already exist, please choose something else'.format(email)  # nopep8
+            )
+        return True
+
+    def check_email_already_in_db(self, email: str) -> bool:
+        """
+        Verify if given email does not already exist in db
+        """
+        return self._session.query(User.email).filter(User.email==email).all() != []  # nopep8
```

Comment:
For information: prefer use `.count()` for readability.

Reply:
Done.